modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -11,12 +12,64 @@ const WORLD_SIZE = 512;
 
 const defaultProps = {
   uniqueIdProperty: {type: 'string', value: ''},
-  highlightedFeatureId: null
+  highlightedFeatureId: null,
+  onTileJSONLoad: {type: 'function', optional: true, value: null, compare: false}
 };
 
 export default class MVTLayer extends TileLayer {
+  initializeState() {
+    super.initializeState();
+    this.setState({
+      data: null,
+      tileJSON: null
+    });
+  }
+
+  get isLoaded() {
+    return this.state.data && super.isLoaded;
+  }
+
+  updateState({props, oldProps, context, changeFlags}) {
+    if (changeFlags.dataChanged) {
+      this._updateTileData({props});
+    }
+    super.updateState({props, oldProps, context, changeFlags});
+  }
+
+  async _updateTileData({props}) {
+    const {onTileJSONLoad} = this.props;
+    let {data} = props;
+    let tileJSON = null;
+
+    if (typeof data === 'string' && !data.match(/\{ *([\w_-]+) *\}/g)) {
```

Comment:
I don't think we support spaces in the template: https://github.com/visgl/deck.gl/blob/8.3-release/modules/geo-layers/src/tile-layer/utils.js#L42-L46